test/integration/replicaschedulingpreference_test.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2018 The Federation v2 Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	fedschedulingv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/federatedscheduling/v1alpha1"
+	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/federation/v1alpha1"
+	"github.com/kubernetes-sigs/federation-v2/pkg/client/clientset_generated/clientset"
+	"github.com/kubernetes-sigs/federation-v2/test/common"
+	"github.com/kubernetes-sigs/federation-v2/test/integration/framework"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// TestReplicaSchedulingPreference validates basic replica scheduling preference calculations.
+var TestReplicaSchedulingPreference = func(t *testing.T) {
+	t.Parallel()
+	tl := framework.NewIntegrationLogger(t)
+
+	controllerFixture, fedClient := initRSPTest(tl, FedFixture)
+	defer controllerFixture.TearDown(tl)
+
+	clusters := getClusterNames(fedClient)
+	if len(clusters) != 2 {
+		tl.Fatalf("Expected two clusters to be part of Federation Fixture setup")
```

Comment:
(No action required) Is failing preferable to skipping here?

It seems strange to me that these tests effectively require all other tests to have 2 member clusters.  Again, this is fine in the near term but I think any test that has specific requirements should probably be deploying its own fixture rather than constraining fixture for others.

Reply:
Actually, I did intend to and deploy a different fixture as part of these tests, but did meet an issue because of how apiserver init code is setup (as per the autogenerated setup code in this repo), the root of which is explained [here](https://github.com/kubernetes-sigs/federation-v2/pull/46#issuecomment-391401574). Two fixtures do not work out fine within the same test context. Perhaps some mechanism to have different test binaries per test setup is the workaround to this problem, but did not seem like necessary at this point of time.

Reply:
That sounds like a point for moving to kubebuilder and CRDs, then, since that would remove the need for an autogenerated apiserver.